config/initializers/saml_idp.rb

```diff
@@ -29,7 +29,7 @@
   # Name ID
   config.name_id.formats =
     {
-      persistent: -> (principal) { principal.uuid },
+      persistent: -> (principal) { principal.last_identity.uuid },
```

Comment:
Using last_identity here seems wrong.  Don't we want to refer to specific identity.  Perhaps the principal here needs to be changed to the identity since the User itself doesn't know which SP/identity is in use.